docs/source/specs/typespec/main.tsp

```diff
@@ -0,0 +1,170 @@
+import "@typespec/http";
+import "@typespec/rest";
+import "@typespec/openapi3";
+import "@typespec/versioning";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+
+@service({title:"Role Based Access Control for Kessel Project"})
+
+@server("https://localhost:{port}/{basePath}", "Development Server", { basePath: string="api/rbac/v2", port: string="8000" })
+@server("https://console.stage.redhat.com/{basePath}", "Stage Server", { basePath: string="api/rbac/v2" })
+@server("https://console.redhat.com/{basePath}", "Production Server", { basePath: string="api/rbac/v2" })
+@versioned(Versions)
+
+namespace KesselRBAC;
+
+enum Versions{
+    v2,
+}
+
+model Timestamps {
+    @doc("2019-01-21T17:32:28Z")
+    created: offsetDateTime;
+    @doc("2019-01-21T17:32:28Z")
+    modified: offsetDateTime;
+}
+
+@format("uuid")
+scalar UUID extends string;
+
+namespace Problems {
+    enum ProblemType {
+        InsufficientPermission:"http://project-kessel.github.io",
+    }
+
+    @error
+    model ProblemDetails<Status extends valueof int16> {
+        @header("content-type") contentType: "application/problem+json";
+        type?: ProblemType;
+    
+        @statusCode _: typeof Status;
+        status?: typeof Status;
+        
+        title?: string;
+        detail?: string;
+
+        @format("uri")
+        instance?: string;
+    }
+  
+    @error
+    model Problem403 extends ProblemDetails<Status = 403>{
+        type: ProblemType.InsufficientPermission,
+        title: "You do not have permission to perform this action.";
+    }
+
+    alias Problem401 = ProblemDetails<Status = 401>;
+
+    @error
+    model Problem404 extends ProblemDetails<Status = 404>{
+        detail: "Not found"
+    }
+    
+    alias Problem500 = ProblemDetails<Status = 500>;
+
+    @error
+    model Problem400 extends ProblemDetails<Status = 400>{
+        detail: "JSON parse error - Expecting value: line 1 column 1 (char 0)",
+    }
+
+    alias Problem400AlreadyExists = ProblemDetails<Status = 400>;
+    
+    alias CommonProblems = Problem401 | Problem403 | Problem500;
+
+    namespace Workspace {
+        model Problem400WorkspaceNotEmpty extends Problems.ProblemDetails<Status = 400>{
+            title: "Unable to delete due to workspace dependencies",
+        }
+    }
+}
+
+@route("/workspaces")
+@tag("Workspaces")
+@doc("Operations about workspaces")
+namespace Workspaces {    
+    model BasicWorkspace {
+        @doc("Workspace A")
+        name: string = "Workspace A";
+        @doc("Description of Workspace A")
+        description?: string = "Description of Workspace A";
+    }
+
+    model Workspace {
+        @key uuid: UUID;
+        parent?: UUID;
+        ...BasicWorkspace;
+        ...Timestamps;
+    }
+
+    alias CreateWorkspaceRequest = BasicWorkspace;
+
+    model CreateWorkspaceResponse extends BasicWorkspace{
+        @statusCode _: 204;
+    }
+
+    model GetWorkspaceResponse extends BasicWorkspace{
```

Comment:
nit: the request/response objects are not always consistent with whether they are using the HTTP verb or the operation name. I would go with the operation name (so CreateWorkspace*, ReadWorkspace*, UpdateWorkspace*, etc).

Reply:
Done 👍